src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -434,32 +428,27 @@ impl<'b, 'tcx:'b> Resolver<'b, 'tcx> {
         }
 
         let name = xcdef.name;
-        let is_public = xcdef.vis == ty::Visibility::Public || parent.is_trait();
-
-        let mut modifiers = DefModifiers::empty();
-        if is_public {
-            modifiers = modifiers | DefModifiers::PUBLIC;
-        }
-        if parent.is_normal() {
-            modifiers = modifiers | DefModifiers::IMPORTABLE;
-        }
+        let vis = if parent.is_trait() { ty::Visibility::Public } else { xcdef.vis };
```

Comment:
Hmm, is this trick still need with `pub(restricted)`? I guess `xcdef.vis` should just work now, no?